src/drivers/serac.cpp

```diff
@@ -15,113 +15,97 @@
-  app.add_option("--ty, --traction-y", ty, "Cantilever tip traction in the y direction.", true);
-  app.add_option("--tz, --traction-z", tz, "Cantilever tip traction in the z direction.", true);
-  app.add_flag("--slu, --superlu, !--no-slu, !--no-superlu", slu_solver, "Use the SuperLU Solver.");
-  app.add_flag("--gmres, !--no-gmres", gmres_solver, "Use gmres, otherwise minimum residual is used.");
-  app.add_option("--lrel, --linear-relative-tolerance", lin_params.rel_tol, "Relative tolerance for the lienar solve.",
-                 true);
-  app.add_option("--labs, --linear-absolute-tolerance", lin_params.abs_tol, "Absolute tolerance for the linear solve.",
-                 true);
-  app.add_option("--lit, --linear-iterations", lin_params.max_iter, "Maximum iterations for the linear solve.", true);
-  app.add_option("--lpl, --linear-print-level", lin_params.print_level, "Linear print level.", true);
-  app.add_option("--nrel, --newton-relative-tolerance", nonlin_params.rel_tol,
-                 "Relative tolerance for the Newton solve.", true);
-  app.add_option("--nabs, --newton-absolute-tolerance", nonlin_params.abs_tol,
-                 "Absolute tolerance for the Newton solve.", true);
-  app.add_option("--nit, --newton-iterations", nonlin_params.max_iter, "Maximum iterations for the Newton solve.",
-                 true);
-  app.add_option("--npl, --newton-print-level", nonlin_params.print_level, "Newton print level.", true);
-  app.add_option("--dt, --time-step", dt, "Time step.", true);
-
-  // Parse the arguments and check if they are good
-  try {
-    app.parse(argc, argv);
-  } catch (const CLI::ParseError& e) {
-    serac::logger::flush();
-    auto err_msg = (e.get_name() == "CallForHelp") ? app.help() : CLI::FailureMessage::simple(&app, e);
-    SLIC_ERROR_ROOT(rank, err_msg);
-    serac::exitGracefully();
-  }
+int main(int argc, char* argv[])
+{
+  auto [num_procs, rank] = serac::initialize(argc, argv);
 
-  auto config_msg = app.config_to_str(true, true);
-  SLIC_INFO_ROOT(rank, config_msg);
+  // Handle Command line
+  auto cli_opts = serac::cli::defineAndParse(argc, argv, rank);
+  serac::cli::printGiven(cli_opts, rank);
 
-  auto mesh = serac::buildMeshFromFile(mesh_file, ser_ref_levels, par_ref_levels);
+  // Read input file
+  std::string input_file_path = "";
+  auto        search          = cli_opts->find("input_file");
+  if (search != cli_opts->end()) input_file_path = search->second;
 
-  int dim = mesh->Dimension();
+  // Create DataStore
+  auto datastore = std::make_shared<axom::sidre::DataStore>();
+
+  // Initialize Inlet and read input file
+  auto inlet = serac::input::initialize(datastore, input_file_path);
+  serac::defineInputFileSchema(inlet, rank);
+
+  // Save input values to file
+  datastore->getRoot()->save("serac_input", "json");
+
+  // serial and parallel refinement levels
+  int ser_ref_levels;
+  inlet->get("ser_ref_levels", ser_ref_levels);  // has default value
```

Comment:
Normally you would check if this was found (Inlet::get returns a boolean), but most of these either have default values or are required in the input file.  I didn't really want to put a bunch of boiler plate code around all these calls so I added a comment why they should not fail.  Thoughts?

Reply:
I don't have a problem with this, but additionally perhaps inlet could have a mode of operation where it aborts the program if a variable is not found and has no default, i.e. one mode where the user should be checking the return values, and another where the user wants to "assert" that all `get`s succeed.

Reply:
It will be common that most codes will have optional values in their input file.  Possibly another idea is to add an optional parameter to `get` that makes it fail if not found.